src/client.rs

```diff
@@ -0,0 +1,557 @@
+    /// Uses the given encoded filter to filter transactions. Note: the actual subscription takes place in
+    /// the background. It will automatically retry every 2s if the stream fails.
+    pub async fn subscribe_new_raw_transactions(
+        &self,
+        filter: Option<Vec<u8>>,
+    ) -> impl Stream<Item = (Address, Bytes)> {
+        let f = match filter {
+            Some(encoded_filter) => TxFilter {
+                encoded: encoded_filter,
+            },
+            None => TxFilter { encoded: vec![] },
+        };
+
+        let key = self.key.clone();
+
+        let mut req = Request::new(f.clone());
+        append_metadata(&mut req, &key);
+
+        let mut client = self.client.clone();
+
+        let (tx, rx) = mpsc::unbounded_channel();
+
+        tokio::spawn(async move {
+            loop {
+                let mut req = Request::new(f.clone());
+                append_metadata(&mut req, &key);
+
+                let mut stream = match client.subscribe_new_txs_v2(req).await {
+                    Ok(stream) => stream.into_inner(),
+                    Err(e) => {
+                        tracing::error!(error = ?e, "Error in transaction stream, retrying...");
+                        tokio::time::sleep(Duration::from_secs(2)).await;
+                        continue;
+                    }
+                };
+
+                tracing::info!("Transaction stream established");
+
+                while let Some(item) = stream.next().await {
+                    match item {
+                        Ok(transaction) => {
+                            let sender = match Address::try_from(transaction.sender.as_slice()) {
+                                Ok(sender) => sender,
+                                Err(e) => {
+                                    tracing::error!(error = ?e, "Error deserializing sender");
+                                    continue;
+                                }
+                            };
+                            let _ = tx.send((sender, transaction.rlp_transaction.into()));
+                        }
+                        Err(e) => {
+                            tracing::error!(error = ?e, "Error in transaction stream, retrying...");
+                            // If we get an error, we set the inner stream to None and break out of the loop.
+                            // Next iteration will retry the stream.
+                            break;
+                        }
+                    }
+                }
+            }
+        });
+
+        UnboundedReceiverStream::new(rx)
+    }
+
+    /// Subscribes to new execution payloads, returning a [`Stream`] of [`ExecutionPayload`].
+    /// Note: the actual subscription takes place in the background.
+    /// It will automatically retry every 2s if the stream fails.
+    pub async fn subscribe_new_execution_payloads(&self) -> impl Stream<Item = ExecutionPayload> {
```

Comment:
This should return a more user friendly execution payload type, maybe something from the alloy-rpc packages